src/main/java/com/limechain/transaction/TransactionProcessor.java

```diff
@@ -118,12 +121,22 @@ private byte[] processTransaction(Extrinsic extrinsic, PeerId peerId) {
             return transactionState.addToPool(validTransaction);
         }
 
-        return transactionState.shouldAddToQueue(validTransaction)
+        var result = transactionState.shouldAddToQueue(validTransaction)
                 ? transactionState.pushTransaction(validTransaction)
                 : transactionState.addToPool(validTransaction);
+
+        if (response.getValidity() != null && response.getValidity().getPropagate()) {
+            messageCoordinator.sendTransactionMessageExcludingPeer(
+                    validTransaction.getExtrinsic(),
+                    validTransaction.getPeersToIgnore()
```

Comment:
Nothing to change for this PR. I just want to leave a comment regarding `peersToIgnore`. Imo not the best solution to use. It would be better to keep track of all incoming messages inside of each peer in some way instead. In this way we can have a centralized place to keep track of all messages received by peers (block announce, transaction, etc).